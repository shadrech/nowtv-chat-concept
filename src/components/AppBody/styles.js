import styled from "react-emotion";

export const BodyWrapper = styled('div')`
  background: red;
  display: flex;
  position: relative;
  width: 100%;
  height: calc(100vh - 6.7rem);
`;

export const ChatLogWrapper = styled('div')`
  flex-grow: 3;
  background: #e5ecef;
  padding: 4rem 5rem;
  overflow-y: scroll;
`;