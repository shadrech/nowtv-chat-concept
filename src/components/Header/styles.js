import styled from "react-emotion";
import {ReactComponent as Icon} from "../../assets/logo.svg";

export const HeaderWrapper = styled('div')`
  width: 100%;
  height: 3.5rem;
  background: #1F2B49;
  display: flex;
  align-items: center;
  justify-content: center;
  position: relative;
`;

export const Logo = styled(Icon)`
  height: 1.5rem;
`;

export const Menu = styled('div')`
  position: absolute;
  right: 1rem;
  height: 100%;
`;

export const MenuItem = styled('div')`
  position: relative;
  display: flex;
  justify-content: space-between;
  align-items: center;
  font-family: Helvetica;
  height: 100%;
  padding: 0 10px;
  &:hover {
    background: rgba(255,255,255,0.1);
    cursor: pointer;
  }
`;

export const ItemText = styled('p')`
  color: white;
  font-size: 1rem;
`;