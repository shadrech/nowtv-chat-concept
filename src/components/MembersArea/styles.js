import styled from "react-emotion";

export const MembersWrapper = styled('div')`
  background: #eff3f5;
  width: 20rem;
`;

export const TopSection = styled('div')`
  position: relative;
  padding: 1rem 0;
  box-sizing: border-box;
  height: 12rem;
  width: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: space-around;
  &:after {
    content: "";
    height: 1px;
    width: 70%;
    position: absolute;
    left: 15%;
    bottom: 0;
    background: #283765;
  }
`;

export const Avatar = styled('div')`
  background-image: url(${props => props.avatar});
  background-size: contain;
  width: 5rem;
  height: 5rem;
`;

export const NameHeading = styled('p')`
  font-size: 1rem;
  font-family: Helvetica, sans-serif;
  font-weight: bold;
  color: #2E3D69;
  margin: 0.1rem 0;
`;

export const DetailsHeading = styled(NameHeading)`
  font-weight: normal;
  font-size: 0.8rem;
`;

export const MiddleSection = styled('div')`
  height: calc(100% - ${props => props.activeMember ? "15rem" : "3rem"});
  overflow-y: scroll;
  padding: 1rem;
  box-sizing: border-box;
`;

export const BottomSection = styled('div')`
  position: relative;
  height: 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  &:after {
    content: "";
    height: 1px;
    width: 70%;
    position: absolute;
    left: 15%;
    top: 0;
    background: #283765;
  }
  p {
    &:hover {
      text-decoration: underline;
      cursor: pointer;
    }
  }
`;