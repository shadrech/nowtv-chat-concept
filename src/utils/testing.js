export const dummyStateData = {
    members: [{
      avatar:null,
      email:"abakera@craigslist.org",
      firstName:"Amanda",
      id:"73f30d93-f87b-45ea-bda5-60fa4b3716b5",
      ip:"82.87.241.19",
      lastName:"Baker",
      messageCount:3
    }],
    messages: [{
      avatar:"http://dummyimage.com/100x100.bmp/5fa2dd/ffffff",
      email:"lowensm@earthlink.net",
      fullName:"Larry Owens",
      message:"Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.",
      messageId:"b03569ae-ccbf-4975-8040-4daba638b407",
      timestamp:"2016-11-09T05:04:58Z",
      userId:"16373df5-da0a-4074-8295-f916b94269f4"
    }]
  };