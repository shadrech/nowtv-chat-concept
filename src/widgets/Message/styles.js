import styled, {css} from "react-emotion"

export const MessageWrapper = styled.div`
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: ${props => props.alignment === "right" ? "flex-end" : "flex-start"};
  margin: 1.25rem 0;
  font-family: sans-serif;
  position: relative;
`;

export const TextWrapper = styled.div`
  background: ${props => props.alignment === "right" ? "#EE5C26" : "white"};
  color: ${props => props.alignment === "right" ? "white" : "#2E3D69"};
  padding: 1rem;
  border-radius: 0.2rem;
  position: relative;
  order: ${props => props.alignment === "right" ? 1 : 2};
  &:after {
    content: "";
    position: absolute;
    bottom: 0;
    height: 0;
    width: 0;
    border-bottom: 1rem solid ${props => props.alignment === "right" ? "#EE5C26" : "white"};
    ${props => props.alignment === "right" ? css`
      right: -0.9rem;
      border-right: 1rem solid transparent;
    ` : css`
      left: -0.9rem;
      border-left: 1.5rem solid transparent;
    `}
    
  }
`;

export const EmailWrapper = styled.figure`
  position: absolute;
  background: #1F2B49;
  padding: 0.5rem;
  display: flex;
  font-size: 0.75rem;
  color: white;
  border-radius: 0.2rem;
  top: -0.5rem;
`;

export const AvatarWrapper = styled.div`
  order: ${props => props.alignment === "right" ? 2 : 1};
  ${props => props.alignment === "right" ? css`
    margin-left: 1rem;
  `: css`
    margin-right: 1rem;
  `}
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  position: relative;
  align-self: flex-end;
`;

export const Time = styled.span`
  font-size: 0.7rem;
  position: absolute;
  bottom: -1.2rem;
  width: 7rem;
`;

export const Avatar = styled.div`
  background-image: url(${props => props.avatar});
  background-position: center;
  background-size: contain;
  width: 3rem;
  height: 3rem;
  border-radius: 1.5rem;
`;