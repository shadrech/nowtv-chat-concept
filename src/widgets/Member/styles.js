import styled, { css } from "react-emotion";
import { Avatar as StandardAvatar } from "../Message/styles";

export const MemberWrapper = styled('div')`
  color: #000;
  margin: 1rem 0;
  background: white;
  border-radius: 0.2rem;
  padding: 0.5rem;
  display: flex;
  &:hover {
    cursor: pointer;
  }
`;

export const Avatar = styled(StandardAvatar)`
  width: 3.5rem;
  height: 3.5rem;
  border-radius: 1.75rem;
`;

export const MemberDetails = styled('div')`
  padding: 0 1rem;
  box-sizing: border-box;
`;

export const Name = styled('p')`
  font-size: 0.9rem;
  font-weight: bold;
  font-family: Helvetica, sans-serif, Arial;
  margin: 0.5rem 0;
`;

export const Text = styled(Name)`
  font-size: 0.75rem;
  font-weight: lighter;
`;