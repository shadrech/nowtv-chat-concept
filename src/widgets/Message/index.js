import React from "react";
import PropType from "prop-types";
import moment from "moment";

import { MessageWrapper, TextWrapper, AvatarWrapper, Avatar, Time } from "./styles";

const formats = {
  sameDay: '[Today at] H:mm',
  nextDay: '[Tomorrow at ] H:mm',
  lastDay: '[Yesterday at ] H:mm',
  sameElse: 'DD/MM/YYYY [at] H:mm'
};

const Message = ({message: {message, avatar, timestamp, email}, alignment}) => (
  <MessageWrapper alignment={alignment}>
    <TextWrapper alignment={alignment}>{message}</TextWrapper>
    <AvatarWrapper alignment={alignment} email={email}>
      <Avatar avatar={avatar} />
      <Time>{moment(timestamp).calendar(null, formats)}</Time>
    </AvatarWrapper>
  </MessageWrapper>
);

Message.propTypes = {
  message: PropType.object,
  alignment: PropType.oneOf(["right", "left"])
};

export default Message;
