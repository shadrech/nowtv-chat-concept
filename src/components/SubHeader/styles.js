import styled from "react-emotion";

import { MenuItem } from "../Header/styles";

export const SubHeaderWrapper = styled.div`
  height: 3.2rem;
  padding: 0 15rem;
  background: #283765;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
`;

export const SubMenuItem = styled(MenuItem)`
  &:after {
    content: "";
    position: absolute;
    bottom: 0;
    left: 0;
    width: 100%;
    height: 0.4rem;
    opacity: ${props => props.active ? 1 : 0};
  }
  &:nth-child(1) {
    &:after {
      background: #e72f8d;
    }
  }
  &:nth-child(2) {
    &:after {
      background: #653796;
    }
  }
  &:nth-child(3) {
    &:after {
      background: #4fade7;
    }
  }
  &:nth-child(4) {
    &:after {
      background: #37ac47;
    }
  }
  &:nth-child(5) {
    &:after {
      background: #5926c0;
    }
  }
  &:nth-child(6) {
    &:after {
      background: #cc091c;
    }
  }

  &:hover {
    background: none;
    &:after {
      opacity: 1;
    }
  }

  p {
    letter-spacing: 0.05rem;
    font-size: 0.9rem;
  }
`;