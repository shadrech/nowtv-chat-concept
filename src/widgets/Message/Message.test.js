import React from "react";
import {mount, shallow} from "enzyme";
import toJson from "enzyme-to-json";
import Message from "./";
import { TextWrapper, AvatarWrapper } from "./styles";
import { dummyStateData } from "../../utils/testing";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion))

describe("<Message />", () => {
  const {messages} = dummyStateData;
  const wrapper = mount(<Message message={messages[0]} alignment="right" />);
  test("should render message in <TextWrapper>", () => {
    expect(wrapper.find(TextWrapper).text()).toBe("Integer a nibh. In quis justo. Maecenas rhoncus aliquam lacus.");
  });
})