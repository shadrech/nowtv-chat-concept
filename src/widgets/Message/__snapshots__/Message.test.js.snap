// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render email on hover 1`] = `
.emotion-0 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-left: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-0:before {
  content: "lowensm@earthlink.net";
  position: absolute;
  background: #1F2B49;
  padding: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  color: white;
  border-radius: 0.2rem;
  cursor: pointer;
  top: -0.5rem;
  opacity: 0;
  -webkit-transition: opacity 0.5s ease;
  transition: opacity 0.5s ease;
}

.emotion-0:hover:before {
  opacity: 1;
}

<Styled(div)
  alignment="right"
  email="lowensm@earthlink.net"
>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;

exports[`<Message /> should render email on hover 2`] = `
.emotion-0 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-left: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
}

.emotion-0:before {
  content: "lowensm@earthlink.net";
  position: absolute;
  background: #1F2B49;
  padding: 0.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 0.75rem;
  color: white;
  border-radius: 0.2rem;
  cursor: pointer;
  top: -0.5rem;
  opacity: 0;
  -webkit-transition: opacity 0.5s ease;
  transition: opacity 0.5s ease;
}

.emotion-0:hover:before {
  opacity: 1;
}

<Styled(div)
  alignment="right"
  email="lowensm@earthlink.net"
>
  <div
    className="emotion-0"
  />
</Styled(div)>
`;
